src/stream/src/executor/managed_state/top_n/top_n_state_new.rs

```diff
@@ -175,6 +176,52 @@ impl<S: StateStore> ManagedTopNStateNew<S> {
         Ok(())
     }
 
+    pub async fn init_topn_cache(
+        &self,
+        pk_prefix: Option<&Row>,
+        topn_cache: &mut TopNCache,
+        epoch: u64,
+    ) -> StreamExecutorResult<()> {
+        assert!(topn_cache.low.is_empty());
+        assert!(topn_cache.middle.is_empty());
+        assert!(topn_cache.high.is_empty());
+
+        let state_table_iter = iter_state_table(&self.state_table, epoch, pk_prefix).await?;
+        pin_mut!(state_table_iter);
+        if topn_cache.offset > 0 {
+            while let Some(item) = state_table_iter.next().await {
```

Comment:
Could you reuse the `fill_cache()` method to populate these caches(e.g. refactor the `start_key` to an `Option`)?  These repeated while loop is kind of verbose.

Reply:
`init_topn_cache()` can populate these caches by calling `iter_state_table()` once, while reusing `fill_cache()` need three times.

Reply:
Got it.